lib/encryptor.rb

```diff
@@ -0,0 +1,50 @@
+require_relative './cryptor'
+require_relative './key'
+require_relative './offset'
+
+class Encryptor < Cryptor
+
+  attr_reader :encrypt_key, :encrypt_offset, :encrypt_shift
+
+  def initialize(message, key = '', date = '') 
+    super(message, key, date) 
+    @encrypt_key = Key.new(key)
+    @encrypt_offset = (date == '' ? Offset.new : Offset.new(date))
+    @encrypt_shift = [@encrypt_key.make_keys, @encrypt_offset.make_offsets].transpose.map{|a| a.sum} 
+  end
+
+  def message_in_alphabet_positions
+    alphabet = ("a".."z").to_a << " "
+    @message.split('').map {|letter| alphabet.find_index(letter)}
+  end
+ 
+  def message_shifts
+    shifts = []
+    (@message.length / @encrypt_shift.length).times {shifts << @encrypt_shift} 
+    shifts << @encrypt_shift[0..(@message.length % @encrypt_shift.length)-1]
+    shifts = shifts.flatten
+    message_in_alphabet_positions.each_with_index do |position, index| 
+      shifts[index] = 0  if position.nil?
```

Comment:
replaces the elements in `shifts` corresponding to positions in the array of the original message where characters that are not part of the alphabet or space appear (!?".,..)